### PR TITLE
Support for external contracts (provided and consumed) for ACI

### PR DESCRIPTION
**Pull request by saramach**
contivmodel changes for external contracts support. Please see related changes to netplugin and aci-gw repositories.
A new object is created to represent external contracts. The epg objects can refer to these objects during EPG create/update.

### CODE REVIEW

File: extContractsGroup.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+	"name": "contivModel",
+	"objects": [
+		{
+			"name": "extContractsGroup",
+			"type": "object",
+			"key": [ "contractsGroupName" ],
```
Comment (jojimt) [inline extContractsGroup.json]: I think we want the key to be tenant, contractsGroupName. @jainvipin what do you think?

Comment (jainvipin) [inline extContractsGroup.json]: this would mean contract group name is global... 
What are the options? It can be qualified by tenant or be left as is.

Comment (jojimt) [inline extContractsGroup.json]: Contracts typically have tenant scope -- the tenant common is the only exception. May be better to keep the scope consistent.

Comment (saramach) [inline extContractsGroup.json]: Resubmitted after adding tenant as part of the key

File: endpointGroup.json [hunk 2]
```diff
@@ -34,15 +34,35 @@
...
 			},
 			"link-sets": {
 				"services": {
 					"ref": "service"
 				},
 				"policies": {
 					"ref": "policy"
-				}
+				},
+                                "consExtContractsGrps": {
```
Comment (jojimt) [inline endpointGroup.json]: I think we only need one array because we have the type inside the contractGroup. Sorry, I missed this part earlier.

**Comment by jojimt** [comment]
Looks good other than the one comment.

File: extContractsGroup.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                "objects": [
+                {
+                        "name": "extContractsGroup",
+                        "type": "object",
+                        "key": [ "tenantName", "contractsGroupName" ],
+                        "properties": {
+                                "tenantName": {
+                                        "type": "string",
+                                        "title": "Tenant name",
+                                        "description": "Tenant name",
```
Comment (shaleman) [inline extContractsGroup.json]: Can you please add validators(format field) to this property?
Please see endpointGroup.json for how to define `format` regex expressions

File: extContractsGroup.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                                        "showSummary": true
+                                },
+                                "contractsGroupName": {
+                                        "type": "string",
+                                        "description": "Contracts group name",
+                                        "title": "Contracts group name",
+                                        "showSummary": true
```
Comment (shaleman) [inline extContractsGroup.json]: Please add `format` validator

File: extContractsGroup.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                                },
+                                "contractsType": {
+                                        "type": "string",
+                                        "description": "Contracts type (Provided -or- Consumed)",
+                                        "title": "Contracts type",
+                                        "showSummary": true
```
Comment (shaleman) [inline extContractsGroup.json:29]: Please add a validator to allow only `provided` or 'consumed` option

**Comment by shaleman** [comment]
LGTM other than the validators.
We can merge this once we add the validator

**Comment by shaleman** [comment]
LGTM

**Pull request merged**